spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkAppenderFactory.java

```diff
@@ -48,6 +48,10 @@
 import org.apache.spark.sql.types.StructType;
 import org.apache.spark.unsafe.types.UTF8String;
 
+/**
+ * @deprecated since 1.7.0, will be removed in 1.8.0; use {@link SparkFileWriterFactory} instead.
+ */
+@Deprecated
 class SparkAppenderFactory implements FileAppenderFactory<InternalRow> {
```

Comment:
While working on partition stats, I came across `FileWriterFactory` and `AppenderFactory`. After digging a bit, found that Spark was using `AppenderFactory` before but now uses `FileWriterFactory`.  But the implementation of `AppenderFactory` still exist and confusing for users. Hence, deprecating it before removal in the next version.

Reply:
Also, this class is not used by the Spark engine integration. Hence, we can safely remove it. 

Looking more into it, we might have lost metrics related function in `FileWriterFactory` that was present in `AppenderFactory`. Maybe I can work on it in the future for `FileWriterFactory`.